specification/appplatform/resource-manager/Microsoft.AppPlatform/preview/2023-07-01-preview/appplatform.json

```diff
@@ -0,0 +1,15096 @@
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "ConfigurationServices_List": {
+            "$ref": "./examples/ConfigurationServices_List.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.AppPlatform/Spring/{serviceName}/configurationServices/{configurationServiceName}/validate": {
+      "post": {
+        "tags": [
+          "AppPlatform"
+        ],
+        "description": "Check if the Application Configuration Service settings are valid.",
+        "operationId": "ConfigurationServices_Validate",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ServiceNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ConfigurationServiceNameParameter"
+          },
+          {
+            "name": "settings",
+            "in": "body",
+            "description": "Application Configuration Service settings to be validated",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/ConfigurationServiceSettings"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success. The response describes the validation result of Application Configuration Service.",
+            "schema": {
+              "$ref": "#/definitions/ConfigurationServiceSettingsValidateResult"
+            }
+          },
+          "202": {
+            "description": "Accepted. The response indicates the ConfigurationServiceSettings is validating.",
+            "schema": {
+              "$ref": "#/definitions/ConfigurationServiceSettingsValidateResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "location"
+        },
+        "x-ms-examples": {
+          "ConfigurationServices_Validate": {
+            "$ref": "./examples/ConfigurationServices_Validate.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.AppPlatform/Spring/{serviceName}/configurationServices/{configurationServiceName}/validateResource": {
```

Comment:
This API validates a configuration service, but it's under a configuration service resource? Would you please clarify how it works?

Reply:
@ms-henglu The validation may involve the configuration service's property, like credential, which requires the configuration service to exist first.

Reply:
How about passing the credential and other required configuration service's property to the validate API?

The current workflow looks strange, because
1. Create the configuration service with credential and other required properties 
2. Call the Validate API for additional properties
3. Update the configuration service?

Reply:
@ms-henglu @TimLovellSmith Thanks for the comment. Our service will also verify the property when creating or updating.  The new API is designed for an existing resource for customers to do that before seriously updating the resource. The separate API can respond faster and won't have any side effects to the resource itself. The behavior is unified in our service.

Reply:
Got it, please do consider improve it in the future.